site/src/hooks/usePrefersReduceMotion.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react'
+
+const query = '(prefers-reduced-motion: no-preference)'
+
+function getPrefersReducedMotion() {
+  if (typeof window === 'undefined') {
+    return false
+  }
+
+  const mediaQueryList = window.matchMedia(query)
+
+  return !mediaQueryList.matches
+}
+
+export function usePrefersReducedMotion() {
+  const [prefersReducedMotion, setPrefersReducedMotion] = useState(getPrefersReducedMotion)
+
+  useEffect(() => {
+    const mediaQueryList = window.matchMedia('(prefers-reduced-motion: no-preference)')
```

Comment:
i think that should be lazy inited in scope and singletoned like? Because now every element create matchmedia and listen it. I think that we could produce only one window matchMedia that will be enabled and it would be nice for it?

Reply:
It's optional here just FYI

Reply:
You are right, but bottleneck here was reactivity and updating all components when user changes mode without reloading the page.
Anyway it could be done via context, since `prefersReducedMotion` is primitive. What do you think?